types/testing-library__jest-dom/index.d.ts

```diff
@@ -251,6 +251,48 @@ declare namespace jest {
          * [testing-library/jest-dom#tohaveclass](https:github.com/testing-library/jest-dom#tohaveclass)
          */
         toHaveClass(...classNames: string[]): R;
+        /**
+         * @description
+         * This allows you to check whether the given form element has the specified displayed value (the one the
+         * end user will see). It accepts <input>, <select> and <textarea> elements with the exception of <input type="checkbox">
+         * and <input type="radio">, which can be meaningfully matched only using toBeChecked or toHaveFormValues.
+         * @example
+         * <label for="input-example">First name</label>
+         * <input type="text" id="input-example" value="Luca" />
+         *
+         * <label for="textarea-example">Description</label>
+         * <textarea id="textarea-example">An example description here.</textarea>
+         *
+         * <label for="single-select-example">Fruit</label>
+         * <select id="single-select-example">
+         *   <option value="">Select a fruit...</option>
+         *   <option value="banana">Banana</option>
+         *   <option value="ananas">Ananas</option>
+         *   <option value="avocado">Avocado</option>
+         * </select>
+         *
+         * <label for="mutiple-select-example">Fruits</label>
+         * <select id="multiple-select-example" multiple>
+         *   <option value="">Select a fruit...</option>
+         *   <option value="banana" selected>Banana</option>
+         *   <option value="ananas">Ananas</option>
+         *   <option value="avocado" selected>Avocado</option>
+         * </select>
+         *
+         * const input = screen.getByLabelText('First name')
+         * const textarea = screen.getByLabelText('Description')
+         * const selectSingle = screen.getByLabelText('Fruit')
+         * const selectMultiple = screen.getByLabelText('Fruits')
+         *
+         * expect(input).toHaveDisplayValue('Luca')
+         * expect(textarea).toHaveDisplayValue('An example description here.')
+         * expect(selectSingle).toHaveDisplayValue('Select a fruit...')
+         * expect(selectMultiple).toHaveDisplayValue(['Banana', 'Avocado'])
+         *
+         * @see
+         * [testing-library/jest-dom#tohavedisplayvalue](https:github.com/testing-library/jest-dom#tohavedisplayvalue)
+         */
+        toHaveDisplayValue(value: string | string[]): R;
```

Comment:
@berickson1 this is the change that is needed:

```suggestion
        toHaveDisplayValue(value: string | RegExp | (string|RegExp)[]): R;
```

And of course, add a couple of tests for when it receives a `RegExp` and an array of them. Also it can receive an array with a mix of regexps and strings. That'd be nice to test as well:

```ts
expect(element).toHaveDisplayValue(/str/);
expect(element).toHaveDisplayValue([/str1/, 'str2']);
```